pyproject.toml

```diff
@@ -0,0 +1,20 @@
+[tool.mypy]
+python_version = "3.12"
+pretty = true
+disallow_any_generics = true
+enable_error_code = "ignore-without-code"
+extra_checks = true
+follow_imports = "silent"
+strict = true
+warn_redundant_casts = true
+warn_unreachable = true
+warn_unused_ignores = true
```

Comment:
A bunch of these are already enabled anyway due to `strict = true`:

```suggestion
python_version = "3.12"
pretty = true
strict = true
follow_imports = "silent"

# Extra checks that aren't included in --strict
enable_error_code = "ignore-without-code"
extra_checks = true
warn_unreachable = true
```

What's the reason for using `follow_imports = "silent"` rather than using the default setting here?

Reply:
I probably just copied and pasted from some other project :)

Reply:
Cool, I think most of the `follow_imports` options do terrible/unexpected things in some situations, so I'd definitely rather stick with the default here if possible!! If there are specific imports that are causing issues and we need a non-default setting, I'd rather we add config for them individually, e.g. like we do in https://github.com/PyCQA/flake8-pyi/blob/f99a1b1dfe713706b3d413a69dd231b9cd2de676/pyproject.toml#L107-L109